PryntTrimmerView/Classes/Trimmer/PryntTrimmerView.swift

```diff
@@ -172,214 +258,253 @@ public protocol TrimmerViewDelegate: AVAssetTimeSelectorDelegate {
+            updateSelectedPositionTime(stoppedMoving: false, triggeredHandle: triggeredHandle)
         case .changed:
             let translation = gestureRecognizer.translation(in: superView)
             if view == leftHandleView {
                 updateLeftConstraint(with: translation)
             } else if view == rightHandleView {
                 updateRightConstraint(with: translation)
+                
             } else {
                 updateLeftConstraint(with: translation)
                 updateRightConstraint(with: translation)
             }
-            fixHandlesLabelsPositionIfNeeded()
             if let startTime = startTime, view == leftHandleView {
                 seek(to: startTime)
             } else if let endTime = endTime {
                 seek(to: endTime)
             }
-            updateSelectedTime(stoppedMoving: false, triggeredHandle: triggeredHandle)
-
+            trimmerDelegate?.trimmerHandleDidMove(triggerHandle:triggeredHandle)
+            updateSelectedPositionTime(stoppedMoving: false, triggeredHandle: triggeredHandle)
         case .cancelled, .ended, .failed:
-            updateSelectedTime(stoppedMoving: true, triggeredHandle: triggeredHandle)
+            updateSelectedPositionTime(stoppedMoving: true, triggeredHandle: triggeredHandle)
+            leftHandleLabel.isHidden = true
+            rightHandleLabel.isHidden = true
+            changeHandleStateColor(color: mainColor)
         default: break
         }
     }
-
+    
+    public func updateWidth(increment: CGFloat) {
+        guard !(increment < 0 && assetPreview.contentWidth == minWidth ||
+              increment > 0 && assetPreview.contentWidth == maxWidth) else { return }
+        propertiesDidChange(widthIncrement: increment)
+        refreshHandles()
+        layoutSubviews()
+    }
+    
     private func updateLeftConstraint(with translation: CGPoint) {
         let maxConstraint = max(rightHandleView.frame.origin.x - minimumDistanceBetweenHandle, handleWidth)
         let minConstraint = max(rightHandleView.frame.origin.x - maximumDistanceBetweenHandle, handleWidth)
-        var newConstraint = min(max(0, currentLeftConstraint + translation.x), maxConstraint)
+        var newConstraint = min(max(0, currentLeftHandleConstraint + translation.x), maxConstraint)
         if newConstraint < minConstraint {
             newConstraint = minConstraint
         }
-        leftConstraint?.constant = newConstraint
+        
+        if assetPreview.contentWidth > UIScreen.main.bounds.width && assetPreview.contentOffset.x > 0 {
+            var offset: CGFloat = 0
+            if newConstraint < 35  {
+                offset = -25
+            } else if newConstraint > assetPreview.bounds.width - 35 {
+                offset = 25
+            }
+            assetPreview.collectionView.contentOffset.x += offset
+            assetPreview.layoutSubviews()
```

Comment:
[c] do not call layout subviews directly

Reply:
done